test_week4_training_config.py

```
@@ -0,0 +1,45 @@
+
+
+def test_load_training_config_accepts_valid_values(tmp_path):
+    config = load_training_config(write_config(tmp_path))
+
+    assert config.epochs == 5
+    assert config.batch_size == 32
+    assert config.dropout == 0.5
+
+
+def test_load_training_config_rejects_zero_epochs(tmp_path):
+    with pytest.raises(ValueError, match="training.epochs must be at least 1"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)